modelling/src/main/java/org/axonframework/modelling/configuration/DefaultStatefulCommandHandlingModule.java

```diff
@@ -39,7 +41,10 @@
 import static java.util.Objects.requireNonNull;
 
 /**
- * Default implementation of the {@link StatefulCommandHandlingModule}.
+ * Default implementation of the {@link StatefulCommandHandlingModule}. Registers the
+ * {@link HierarchicalStateManagerConfigurationEnhancer} enhancer to the module so that message handlers get access
```

Comment:
Perhaps I am missing this part, but where does the `DefaultStatefulCommandHandlingModule` register the `HierarchicalStateManagerConfigurationEnhancer` exactly?

Reply:
It's loaded via the ServiceLoader mechanism!

Reply:
First I added it hard-coded, but maybe someone wants to turn it off for a module. 
So, with the service loader loading it, users could disable it singularly via the new methods introduced on the `ComponentRegistry`

Reply:
Ah of course...this makes sense now. Thanks for clarifying!